url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/DeviceGraphicsDisplay/KitRobotGraphicsDisplay.java

```diff
@@ -0,0 +1,577 @@
+				kits.add(currentKit);
+				InspectionToGoodConveyor();
+			} else if (command
+					.equals(Constants.KIT_ROBOT_DISPLAY_PICKS_LOCATION1_TO_INSPECTION)) {
+				KitGraphicsDisplay tempKit = new KitGraphicsDisplay(
+						(KitConfig) obj);
+				KitConfig kitConfig = (KitConfig) obj;
+				tempKit.setLocation(kitConfig.getLocation());
+				setKitConfigurations(tempKit, 0);
+				kits.add(currentKit);
+				Location1ToInspectionStand();
+			} else if (command
+					.equals(Constants.KIT_ROBOT_DISPLAY_PICKS_LOCATION2_TO_INSPECTION)) {
+				KitGraphicsDisplay tempKit = new KitGraphicsDisplay(
+						(KitConfig) obj);
+				tempKit.setLocation(location2);
+				setKitConfigurations(tempKit, 0);
+				kits.add(currentKit);
+				Location2ToInspectionStand();
+			} else if (command
+					.equals(Constants.KIT_ROBOT_DISPLAY_PICKS_INSPECTION_TO_LOCATION1)) {
+				KitGraphicsDisplay tempKit = new KitGraphicsDisplay(
+						(KitConfig) obj);
+				tempKit.setLocation(inspectionLocation);
+				setKitConfigurations(tempKit, 1);
+				kits.add(currentKit);
+				InspectionToLocation1();
+			} else if (command
+					.equals(Constants.KIT_ROBOT_DISPLAY_PICKS_INSPECTION_TO_LOCATION2)) {
+				KitGraphicsDisplay tempKit = new KitGraphicsDisplay(
+						(KitConfig) obj);
+				tempKit.setLocation(inspectionLocation);
+				setKitConfigurations(tempKit, 2);
+				kits.add(currentKit);
+				InspectionToLocation2();
+			}
+
+		}
+
+	}
+
+	/**
+	 * rotates the kit robot if you are doing a job
+	 */
+	public void doJob() {
+		if (!jobIsDone) {
+			kitMagicX += this.xStep;
+			kitMagicY += this.yStep;
+			stepCountDown -= 1;
+			if (finalJob) {
+				currentKit.setLocation(new Location(currentKit.getLocation()
+						.getX() + this.xStep, currentKit.getLocation().getY()
+						+ this.yStep));
+			}
+			// System.out.println("currentDegree: " + currentDegree);
+		}
+	}
+
+	/*
+	 * draws the kit robot and kit (non-Javadoc)
+	 * 
+	 * @see
+	 * DeviceGraphicsDisplay.DeviceGraphicsDisplay#draw(javax.swing.JComponent,
+	 * java.awt.Graphics2D)
+	 */
+	public void draw(JComponent c, Graphics2D g) {
+		checkDegrees();
+		doJob();
+		if (!invisibleCloud) {
+			g.drawImage(Constants.KIT_CLOUD_IMAGE, kitMagicX, kitMagicY, null);
+		}
+
+		drawtheKits(c, g);
+		// g.drawImage(Constants.KIT_ROBOT_IMAGE_FLICKER, kitRobotPositionX,
+		// kitRobotPositionY, c);
+		animateRobot(g, c);
+	}
+
+	/*
+	 * draws the kits in the kit robot area
+	 */
+	public void drawtheKits(JComponent c, Graphics2D g) {
+		for (int i = 0; i < kits.size(); i++) {
+			kits.get(i).drawKit(c, g);
+		}
+	}
+
+	public void animateRobot(Graphics2D g, JComponent c) {
+		img = Toolkit.getDefaultToolkit().getImage(
+				Constants.KIT_ROBOT_IMAGE + seq + ".png");
+		if (animateKitRobot) {		
+			if (animCount % 7 == 0) {
+				sequenceIncrease();
+			} else if (animCount < 1) {
+				animCount = 76;
+			}
+
+			animCount--;
+		}
+		g.drawImage(img, kitRobotPositionX, kitRobotPositionY, c);
+	}
+
+	public void sequenceIncrease() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.